recipes/extending-kysely.md

```diff
@@ -7,18 +7,17 @@ Kysely makes this really simple.
 
 The Kysely API is designed around two interfaces [`Expression<T>`](https://koskimas.github.io/kysely/interfaces/Expression.html)
 and [`AliasedExpression<T, A>`](https://koskimas.github.io/kysely/interfaces/AliasedExpression.html).
-Almost every method accepts values that implement these interfaces and most Kysely classes like
-`SelectQueryBuilder` and `RawBuilder<T>` achieve their type magic by implementing them.
+Almost every method accepts values that implement these interfaces and most Kysely internals achieve their "type magic" by implementing them.
 
 Most of the time you can create your helpers using the [sql template tag](https://koskimas.github.io/kysely/interfaces/Sql.html)
-and the `RawBuilder<T>` and `AliasedRawBuilder<T, A>` classes it returns but it's good to first understand how
-the underlying interfaces `Expression<T>` and `AliasedExpression<T, A>` work.
+and the `RawBuilder<T>` and `AliasedRawBuilder<T, A>` class instances it returns, but it's good to first understand how
+the underlying interfaces they implement, `Expression<T>` and `AliasedExpression<T, A>`, work.
 
 ## Expression<T>
 
-[`Expresssion<T>`](https://koskimas.github.io/kysely/interfaces/Expression.html) is a simple interface
-that has a type `T` and a single method `toOperationNode()`. `T` tells Kysely's type system the type of
-the expression. `toOperationNode()` returns instructions on what SQL should be produced once the
+[`Expression<T>`](https://koskimas.github.io/kysely/interfaces/Expression.html) is a simple interface
+that has a type `T` and a single method `toOperationNode()`. `T` tells Kysely's type system the result type of
+of the expression once the expression (or its ancestor) is executed. `toOperationNode()` returns instructions on what SQL should be produced once the
```

Comment:
Not only when it's executed. It's used by `where` and other methods to check against field's (or other expression's) type.